packages/augur-state/src/db/DB.ts

```diff
@@ -146,76 +150,177 @@ export class DB<TBigNumber> {
       }
     }
 
-    this.metaDatabase = new MetaDB(networkId + "-BlockNumberEvents");
+    // TODO Initialize full-text DB
+
+    this.metaDatabase = new MetaDB(this, networkId);
   }
 
   /**
    * Called from SyncableDB constructor once SyncableDB is successfully created.
    * 
-   * @param networkId Network to which the SyncableDB connects
-   * @param eventName Generic event type on which the SyncableDB syncs
-   * @param db dbController that utilizes the SyncableDB
+   * @param {SyncableDB<TBigNumber>} db dbController that utilizes the SyncableDB
    */
-  public notifySyncableDBAdded(networkId: number, eventName: string, db: SyncableDB<TBigNumber>): void {
-    this.syncableDatabases[`${networkId}-${eventName}`] = db;
+  public notifySyncableDBAdded(db: SyncableDB<TBigNumber>): void {
+    this.syncableDatabases[db.dbName] = db;
   }
 
   /**
    * Called from UserSyncableDB constructor once UserSyncableDB is successfully created.
    * 
-   * @param networkId Network to which the UserSyncableDB connects
-   * @param userSpecificEventName User-specific event type on which the UserSyncableDB syncs
-   * @param trackedUser User address for which to sync the user-specific event type
-   * @param db dbController that utilizes the UserSyncableDB
+   * @param {UserSyncableDB<TBigNumber>} db dbController that utilizes the UserSyncableDB
    */
-  public notifyUserSyncableDBAdded(networkId: number, userSpecificEventName: string, trackedUser: string, db: UserSyncableDB<TBigNumber>): void {
-    this.userSyncableDatabases[`${networkId}-${userSpecificEventName}-${trackedUser}`] = db;
+  public notifyUserSyncableDBAdded(db: UserSyncableDB<TBigNumber>): void {
+    this.userSyncableDatabases[db.dbName] = db;
   }
 
   /**
    * Syncs generic events and user-specific events with blockchain and updates MetaDB info.
    * 
-   * @param augur Augur object with which to sync
-   * @param chunkSize Number of blocks to retrieve at a time when syncing logs
-   * @param blockstreamDelay Number of blocks by which blockstream is behind the blockchain
-   * @param uploadBlockNumber Block number at which Augur contracts were originally uploaded to the blockchain
+   * @param {Augur<TBigNumber>} augur Augur object with which to sync
+   * @param {number} chunkSize Number of blocks to retrieve at a time when syncing logs
+   * @param {number} blockstreamDelay Number of blocks by which blockstream is behind the blockchain
+   * @param {number} uploadBlockNumber Block number at which Augur contracts were originally uploaded to the blockchain
    */
   public async sync(augur: Augur<TBigNumber>, chunkSize: number, blockstreamDelay: number, uploadBlockNumber: number): Promise<void> {
-    let sequenceIds: { [eventName: string]: string } = {};
+    // Sync generic event types & user-specific event types
+    let dbSyncPromises = [];
+    for (let eventName of genericEventNames) {
+      let dbName = `${this.networkId}-${eventName}`;
```

Comment:
I recommend creating a separate function for creating these kinds of values, to minimize the risk of later typos causing confusing bugs.

Reply:
Yeah, good idea. I'll add that.